.gitattributes

```diff
@@ -0,0 +1,81 @@
+# Set the default behavior, in case people don't have core.autocrlf set.
+* text=auto
+
+# Files with these extensions are accepted source files inside the ReactOS
+# tree. They should have native line endings of the OS.
+*.acf text
+*.asm text
+*.c text
+*.cc text
+*.cmake text
+*.cpp text
+*.cxx text
+*.h text
+*.hpp text
+*.hxx text
+*.idl text
+*.inc text
+*.inl text
+*.klc text
+*.l text
+*.lds text
+*.md text
+*.rc text
+*.s text
+*.S text
+*.sfd text
+*.spec text
+*.svg text
+*.txt text
+*.TXT text
+*.y text
+
+# Files with these extensions end up in the built ReactOS system, so they
+# need to have CRLF line endings.
+*.bat text eol=crlf
+*.cmd text eol=crlf
+*.exp text eol=crlf
+*.hhc text eol=crlf
+*.hhp text eol=crlf
+*.ini text eol=crlf
+*.INI text eol=crlf
+*.inf text eol=crlf
+*.inf.tpl text eol=crlf
+*.js text eol=crlf
+*.manifest text eol=crlf
+*.mc text eol=crlf
+*.mof text eol=crlf
+*.rgs text eol=crlf
+*.sif text eol=crlf
+*.vbs text eol=crlf
+
+# Files with these extensions are accepted binary files inside the ReactOS
+# tree. Git must not tamper with them at all!
+*.avi binary
+*.bin binary
+*.bmf binary
+*.bmp binary
+*.BMP binary
+*.chm binary
+*.cur binary
+*.dll binary
+*.emf binary
+*.gif binary
+*.ico binary
+*.jpg binary
+*.mp3 binary
+*.nls binary
+*.otf binary
+*.pdn binary
+*.pfb binary
+*.pfm binary
+*.png binary
+*.psd binary
+*.ttc binary
+*.ttf binary
+*.wav binary
+*.xcf binary
+
+# All other extensions not explicitly mentioned here are left for Git to
+# handle automatically.
```

Comment:
I'm not sure I understand your motiviation for leaving out dsp/dsw/vc(x)proj. As far as I know we _do_ have such files in the repo, so we should avoid breaking them.

Reply:
VS does not really care for vc(x)proj afaik, however it can create a mess.

Reply:
Well, my motivation is more like: There shouldn't be any VS project files in the repo anymore when we have CMake to create them automatically. Especially given the fact that they are outdated with every new VS version.
So not adding them to .gitattributes will still check them out with native CRLF line-endings on Windows (as they are detected as text files) and the few people who really need them can use them. But you should never ever modify them under Unix environments, and better get rid of them entirely anyway.

Reply:
I would assume most/all of the VS project files we have are just taken bona fide from third party code. In that case they're there for ease of syncing or for editing that particular library, i.e. should be unmodified and usable, and shouldn't go away long-term.

Reply:
I have one in there, which is used to conveniently test ATL apitest code against microsoft headers....

Reply:
Ok, I can add them with crlf line endings to the list.
But I don't want this to be understood as a recommendation to drop VS project files into our repo.

Reply:
@learn-more: Ok, I get the idea of your .vcxproj file and can add eol=crlf for it. Let's hope MS won't outdate it for a while.
I'm still in favor of not adding a rule for the outdated .vcproj or even .dsp/.dsw files though. If they only exist for syncing third-party code, Git's automatic detection of them as text files should be enough. Otherwise, we have to add rules for each and every README file of that third-party code too..
Furthermore, every decent diff tool these days supports ignoring line ending changes when comparing.